File: Actividad 3 Gráfico de Barras en Consola/Actividad3.java
public class Actividad3 {

    public static void main(String[] args) {

        int[] valores = {20, 50, 80, 40, 60, 10, 70, 30};

        // Encontrar valor máximo
        int maximo = valores[0];

        for (int valor : valores) {
            if (valor > maximo) {
                maximo = valor;
            }
        }

        System.out.println("GRÁFICO DE BARRAS\n");

        for (int i = 0; i < valores.length; i++) {

            // Calcular cantidad de asteriscos
            int asteriscos = (int) Math.round(((double) valores[i] / maximo) * 20);

            // Imprimir índice y valor
            System.out.print("[" + i + "] " + valores[i] + " -> ");

            // Imprimir barra
            for (int j = 0; j < asteriscos; j++) {
                System.out.print("*");
            }

            System.out.println();
        }
    }
}

File: Actividad 3 Gráfico de Barras en Consola/Actividad3.py
import math

valores = [20, 50, 80, 40, 60, 10, 70, 30]

# Encontrar valor máximo
maximo = max(valores)

print("GRÁFICO DE BARRAS\n")

for i in range(len(valores)):

    # Calcular cantidad de asteriscos
    asteriscos = round((valores[i] / maximo) * 20)

    # Crear barra
    barra = "*" * asteriscos

    print(f"[{i}] {valores[i]} -> {barra}")
